code/modules/mob/mob.dm

```diff
@@ -105,6 +105,25 @@
 	else
 		to_chat(src, msg)
 
+/mob/proc/display_output(sound/S, image/I, text, turf/turf_source, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+	//Process sound
+	if(S && can_hear())
+		SEND_SOUND(src, S)
+	//Process icon
+	if(isliving(src))
```

Comment:
instead of checking the type of src override the mob/living/display_output() with ..() and the audiolocation code